src/lseq/mod.rs

```diff
@@ -0,0 +1,155 @@
+/// Contains the implementation of the exponential tree for LSeq
+pub mod ident;
+
+use ident::*;
+use serde::{Deserialize, Serialize};
+
+use crate::traits::CmRDT;
+
+use crate::vclock::{Dot, Actor};
+
+// SiteId can be generalized to any A if there is a way to generate a single invalid actor id at every site
+// Currently we rely on every site using the Id 0 for that purpose.
+
+/// Actor type for LSEQ sites.
+#[derive(Debug, Hash, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct SiteId(u32);
+
+// identifier, clock, site id, element
+type Entry<T> = (Identifier, Dot<SiteId>, T);
+
+/// LSeq tree
+///
+/// An LSeq tree is a CRDT for storing sequences of data (Strings, ordered lists)
+/// Internally it works by viewing each element as the leaf of a giant tree.
+/// The path that leads to a given element is called the 'identifier' of that element
+///
+/// LSeq is very similar to the LOGOOT algorithm for representing sequences. The major change is
+/// that LSeq uses an **exponential** tree to store data. That means that at each level of the tree
+/// the space doubles. This helps prevent growth of identifier sizes.
+//#[derive(Serialize, Deserialize)]
+pub struct LSeq<T> {
+    seq: Vec<Entry< T>>,
+    gen: IdentGen,
+    dot: Dot<SiteId>
```

Comment:
This could be generalized to any `Actor` if we could generate a 'default' actor that is invalid / unused across every site.